[Remove Vuetify from Studio] Channel details in Channels - page layout

vtushar06 opened the pull request:
<!--
 1. Following guidance below, replace …'s with your own words
 2. After saving the PR, tick of completed checklist items
 3. Skip checklist items that are not applicable or not necessary
 4. Delete instruction/comment blocks
-->

## Summary
<!--
 * description of the change
 * manual verification steps performed
 * screenshots if the PR affects the UI
-->
Migrates `ChannelDetailsModal` from Vuetify to Kolibri Design System by creating a new `StudioImmersiveModal` component.

### Changes:
- **Created** `StudioImmersiveModal.vue` - Fullscreen modal using KToolbar and KIconButton (no Vuetify dependencies)
- **Refactored** `ChannelDetailsModal.vue`:
  - Replaced `FullscreenModal` with `StudioImmersiveModal`
  - Replaced Vuetify download menu with `KButton` + `KDropdownMenu`
  - Replaced `LoadingText` with `StudioLargeLoader`
  - Removed all Vuetify components (VCard, VLayout, VSpacer, VBtn, VList, etc.)
  - Preserved `DetailsPanel` unchanged per requirements



### Screenshots:
<img width="1512" height="903" alt="Screenshot 2025-11-03 at 2 50 23 PM" src="https://github.com/user-attachments/assets/dcc812e1-be15-4008-b84c-4399a16ace78" />
<img width="1511" height="896" alt="Screenshot 2025-11-03 at 2 50 10 PM" src="https://github.com/user-attachments/assets/0bea2db7-f72e-4e49-b453-1bba99205b3a" />

<img width="326" height="729" alt="Screenshot 2025-10-27 at 9 21 36 AM" src="https://github.com/user-attachments/assets/c1609a47-b987-4c28-9152-583326318d0e" />

## References
<!--
 * references to related issues and PRs
 * links to mockups or specs for new features
 * links to the diffs for any dependency updates, e.g. in iceqube or the perseus plugin
-->
**Closes:** #5474 
…

## Reviewer guidance
<!--
 * how can a reviewer test these changes?
 * are there any risky areas that deserve extra testing
-->
## Quick Test:
1. `pnpm install && pnpm run devserver:hot`
2. Login as `a@a.com` / `a`
3. Go to Channels > My Channels
4. Click info icon (ⓘ) on any channel
5. Verify: toolbar, close button, ESC key, download dropdown, responsive layout

### Key Test Cases:
- [x] Close button & ESC key work
- [x] Download PDF/CSV functional
- [x] Mobile: full-width button
- [x] RTL: proper spacing and alignment (`pnpm run devserver`)
- [x] Offline alert appears when disconnected

MisRob commented:
@vtushar06 we won't do any changes to user experience unless explicitly mentioned in the issue - it's not the goal of this project, and we have internal processes that would need to precede such decisions. So please stay as close as possible to the current experience.

vtushar06 commented:
okay sure @MisRob, I thought in sake for UX, reverting back the changes to use current experience, thanks for this guidance too.

MisRob commented:
Hi @vtushar06, just following-up - I plan to review this PR next week. I posted few notes on your other PRs and some of them were quite general, so meanwhile you can see if any of them apply here as well.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/channel/__tests__/channelDetailsModal.spec.js:
```diff
@@ -1,99 +1,236 @@
-      await wrapper.vm.load();
-      expect(document.title).toContain(channel.name);
+    await waitFor(() => {
+      expect(screen.getByText(testChannel.name)).toBeInTheDocument();
```
You attempt to test that `'should track analytics on mount'` but this expect doesn't check anything like that. I think it'd be fine to not test analytics though. You can either fix, or remove.

MisRob replied:
You attempt to test that `'should track analytics on mount'` but this expect doesn't check anything like that. I think it'd be fine to not test analytics though. You can either fix, or remove.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/channel/ChannelDetailsModal.vue:
```diff
@@ -187,14 +182,24 @@
+    justify-content: flex-start;
+  }
+
+  @media (max-width: 600px) {
```
We avoid CSS media queries in favor of our layout system - mostly for consistency. Can you use `useKResponsiveWindow`?

- https://design-system.learningequality.org/layout#responsiveness
- https://design-system.learningequality.org/usekresponsivewindow/

MisRob replied:
We avoid CSS media queries in favor of our layout system - mostly for consistency. Can you use `useKResponsiveWindow`?

- https://design-system.learningequality.org/layout#responsiveness
- https://design-system.learningequality.org/usekresponsivewindow/

MisRob commented on contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue:
```diff
@@ -0,0 +1,139 @@
+    right: 0;
+    bottom: 0;
+    left: 0;
+    z-index: 17;
```
Based on the diagram in the _Z-indexes_ section of https://design-system.learningequality.org/styling, will `z-index` 16 work?

Sometimes we run into conflicts with Vuetify though, so it's not possible to follow this guidance all the time until the whole Studio is migrated. Maybe just try and see if you observe any problems and then we can decide.

MisRob replied:
Based on the diagram in the _Z-indexes_ section of https://design-system.learningequality.org/styling, will `z-index` 16 work?

Sometimes we run into conflicts with Vuetify though, so it's not possible to follow this guidance all the time until the whole Studio is migrated. Maybe just try and see if you observe any problems and then we can decide.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue:
```diff
@@ -0,0 +1,139 @@
+      });
+    },
+    methods: {
+      hideHTMLScroll(hidden) {
```
If the `html` element had a custom overflow style before the modal opened, it would be lost when the modal closes due to this method.

Can you explore alternative approaches that are more robust in this regard? There are more techniques to achieve this. I imagine we would use something fairly straightforward and CSS class based. Before you re-implement code, you can share a brief summary of options you're thinking about here and then we can decide.

MisRob replied:
Yes, that sounds good to me :+1:

MisRob commented on contentcuration/contentcuration/frontend/shared/views/StudioPage.vue:
```diff
@@ -31,10 +34,19 @@
     },
     props: {
       offline: { type: Boolean, default: false },
+      marginTop: { type: Number, default: null },
```
```suggestion
      marginTop: { type: Number, default: 104 },
```

Then you could remove `const marginTop = props.marginTop !== null ? props.marginTop : 104;` and just rely on the default.

vtushar06 replied:
I changed the default to 104 and simplified the computed property to always use the marginTop value directly

MisRob commented on contentcuration/contentcuration/frontend/shared/views/channel/ChannelDetailsModal.vue:
```diff
@@ -187,14 +182,24 @@
-    padding-right: 10px;
-    font-style: italic;
-    color: gray;
+  [dir='rtl'] .download-button-container {
```
I appreciated you think of RTL. I would expect the RTLCSS framework we use (https://kolibri-dev.readthedocs.io/en/develop/i18n.html#text-alignment) to take care of switching `flex-start` for `flex-end` automatically though, so this style may not be needed.

However, this may not be possible to preview when running the development server in the hot reload mode. Can you try to preview with just `pnpm run devserver` without `:hot`? If it shows okay in RTL without this style, then you can remove it.

vtushar06 replied:
Tested it - the RTL rule is needed. With `justify-content: flex-end` in the [dir='rtl'] selector, the button correctly appears on the left in RTL mode. RTLCSS handles the transformation properly when the rule is present.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/channel/__tests__/channelDetailsModal.spec.js:
```diff
@@ -1,99 +1,236 @@
-function makeWrapper() {
-  return mount(ChannelDetailsModal, {
-    router,
+const createRouter = () => {
```
Can  be simplified to

```
const createRouter = () => {
  return new VueRouter({
    routes: [
      {
        path: '/',
        children: [
          {
            path: '/channel/:channelId',
          },
        ],
      },
    ],
  });
};
```

and you can also remove `router.push` below on line 78.

MisRob reviewed (changes requested):
Hi @vtushar06, thank you. Looks pretty good overall. I'm leaving some notes and few questions.

In my localhost preview, I noticed these three areas that need to be addressed:

<img width="1427" height="1133" alt="review-channel-layout" src="https://github.com/user-attachments/assets/c056b287-6317-4d8d-a361-6f7545986ad8" />

vtushar06 commented on contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue:
```diff
@@ -0,0 +1,139 @@
+    left: 0;
+    z-index: 17;
+    overflow-y: auto;
+    background-color: white;
```
Removed hardcoded white background-color and now using the KDS surface token via :style binding.

vtushar06 commented on contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue:
```diff
@@ -0,0 +1,139 @@
+        required: false,
+        default: '',
+      },
+      color: {
```
done, removed color and dark props and simplified the toolbar to show black background with white text.

vtushar06 commented on contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue:
```diff
@@ -0,0 +1,139 @@
+
+  .toolbar-title {
+    display: block;
+    max-width: calc(100% - 80px);
```
I've now removed those properties while keeping the margin spacing for proper RTL support. The toolbar title layout should now be handled entirely by KToolbar itself. 
basically I wrote these line when I am handling the margin but didn't removed after testing that.
Thank you for catching this - I'll be more thorough about testing design decisions before implementation.

vtushar06 commented on contentcuration/contentcuration/frontend/shared/views/channel/ChannelDetailsModal.vue:
```diff
@@ -1,57 +1,39 @@
-                :color="$themeTokens.textInverted"
-              />
-            </VBtn>
+    <StudioLargeLoader v-if="loading" />
```
now,I have implemented useKShow with minVisibleTime of 500ms to prevent loader flashing when data loads quickly.

vtushar06 commented on contentcuration/contentcuration/frontend/shared/views/channel/__tests__/channelDetailsModal.spec.js:
```diff
@@ -1,99 +1,236 @@
+      },
+    },
+    mocks: {
+      $analytics: {
```
Thanks for the suggestion, I tested removing the $analytics mock and the analytics calls still work fine without it—they just go through to the real analytics instance. So we can safely remove it and keep the test setup minimal as you mentioned.

vtushar06 commented:
https://github.com/user-attachments/assets/9f457e09-0b38-4c68-9880-debd5c71780c

Hi @MisRob,
Thanks for the thorough review! I've made all the updates:
All the changes follow the principles you suggested. The PR should be much cleaner now!

MisRob commented:
The recording looks good @vtushar06, I will re-review. Can you have a look at why tests are failing?

MisRob commented:
@vtushar06 I don't understand fully what you're trying to say around pre-existing issue, but I don't see a problem with adding test IDs as needed. E.g. `StudioLargeLoader` was added here in this PR. So if you're adding/updating a test that needs to detect the loader with the test ID, you need to put that test ID on `StudioLargerLoader`.

vtushar06 commented:
you're right, my apologies for the confusion. I understand now - since I added StudioLargeLoader in this PR, it's my job to make sure it has the test IDs tests need. I've now added data-testid="loader" to it. Thanks for clarifying.

vtushar06 commented:
Now PR is ready to get final review.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue:
```diff
@@ -0,0 +1,125 @@
+          icon="close"
+          :ariaLabel="$tr('close')"
+          :tooltip="$tr('close')"
+          color="white"
```
`textInverted` token - see KDS docs.

MisRob replied:
`textInverted` token - see KDS docs.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue:
```diff
@@ -0,0 +1,125 @@
+      </template>
+    </KToolbar>
+
+    <StudioOfflineAlert :offset="64" />
```
This is too large offset causing extra space between the toolbar and offline bar.

Also, when offline, _Download channel summary_ button shouldn't be below the offline bar, not even partially.

MisRob replied:
This is too large offset causing extra space between the toolbar and offline bar.

Also, when offline, _Download channel summary_ button shouldn't be hidden below the offline bar, not even partially.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue:
```diff
@@ -0,0 +1,125 @@
+      </template>
+
+      <template #default>
+        <span class="notranslate toolbar-title">
```
`notranslate` is already applied from outside this component. Can be removed.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/channel/__tests__/channelDetailsModal.spec.js:
```diff
@@ -1,99 +1,155 @@
+      expect(screen.getByText(testChannel.name)).toBeInTheDocument();
+    });
+
+    const closeButton = screen.getByLabelText('Close');
```
For querying buttons here and elsewhere, see VTL's `getByRole`

MisRob replied:
For querying buttons here and elsewhere, see VTL's `getByRole`

MisRob commented on contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue:
```diff
@@ -0,0 +1,125 @@
+        <KIconButton
+          icon="close"
+          :ariaLabel="$tr('close')"
+          :tooltip="$tr('close')"
```
Tooltip is misplaced. This is sometimes happening in Studio, we're not yet sure about the cause. Probably best to just remove `tooltip` prop. It wasn't present before anyway.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue:
```diff
@@ -0,0 +1,125 @@
+</script>
+
+
+<style lang="scss" scoped>
```
There are two `<style lang="scss" scoped>` blocks. After fixed, you will need to revisit how `overflow` styles are set, and ensure that there's no extra style, and understand impact of each change on user experience.

When opening a new PR, or before asking _for each new round of re-review_, I suggest you open _the Files changed_ tab, and read each single line carefully - intentionally asking questions (_Is this change needed?_ _What's its real impact?_ and similar), and apply critical thinking. This way, you can improve PRs significantly before requesting reviewers' time on things that you can take care of on your own. Additionally, it serves as training for becoming a good reviewer. I do this with my own work and it helps a lot. The trick is to forget that you implemented it, whys&hows, and try your best to locate problematic areas. Please try it out on this and other PRs that you have open, and see.

vtushar06 replied:
Hi @MisRob, I wanted to clarify something about the two <style> blocks in StudioImmersiveModal.vue - this is actually intentional and working as designed. The first one (scoped) handles the modal's internal layout - things like the fixed positioning and preventing scrolling inside the modal itself. The second one (unscoped) is for the global behavior - when the modal opens, we add a class to the HTML element to prevent the body from scrolling in the background. The layout looks good with no double scrollbars or spacing issues. Let me know if you'd like me to change anything about this approach!

vtushar06 replied:
Also removed overflow-x: hidden from StudioPage's .studio-page-outer - same thing, no impact on layout
Everything works great - modal scrolling behaves correctly and there are no unwanted scrollbars.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/channel/ChannelDetailsModal.vue:
```diff
@@ -1,78 +1,81 @@
+          :text="$tr('downloadButton')"
+          :primary="true"
+          hasDropdown
+          data-test="download-button"
```
Unused attribute. I see some other unused data, such as `class`es. Please check all files and cleanup.

vtushar06 replied:
removed these many unused data ,class="download-button" - unused CSS class
loadError: false - unused data property
this.loadError = true;

MisRob commented on contentcuration/contentcuration/frontend/shared/views/channel/__tests__/channelDetailsModal.spec.js:
```diff
@@ -1,99 +1,155 @@
     });
+  });
+
+  it('should call loadChannel and loadChannelDetails on mount', async () => {
```
One possible improvement that is generally useful to check on is that actions were called with correct parameters, here with the right channel ID.

vtushar06 replied:
now I have changed that to "should call loadChannel and loadChannelDetails on mount with correct channel ID"
and used `expect(mockActions.loadChannel).toHaveBeenCalledTimes(1);` and `expect(mockActions.loadChannelDetails).toHaveBeenCalledTimes(1);`
Please verify that are these looks good?

vtushar06 commented on contentcuration/contentcuration/frontend/shared/views/StudioPage.vue:
```diff
@@ -45,16 +56,14 @@
   .studio-page-inner {
     width: 100%;
-    margin-left: 0;
+    margin: 0 auto;
```
You're absolutely right - I see the issue now. I've removed the margin: 0 auto from StudioPage and it's back to left-aligned by default, so the Settings page works correctly again. The modal will handle any centering it needs from its own styles instead.

MisRob commented on contentcuration/contentcuration/frontend/shared/views/channel/ChannelDetailsModal.vue:
```diff
@@ -1,78 +1,81 @@
+        class="download-button-container"
+        :style="downloadButtonContainerStyle"
+      >
+        <KButton
```
Perhaps around 50px or so? The offline bar needs to be right below the toolbar. So that scrollbar is not visible between the two - right now you can see a little piece of it.

Also, page content needs to be centered.

vtushar06 replied:
I've fixed both issues:
Offline bar offset: Changed to 46px - now sits right below the toolbar with no gap or scrollbar peeking through
Page content centering: Added a centered prop to StudioPage that StudioImmersiveModal passes as true. This applies margin: 0 auto to the inner content from outside the component, so Settings page stays left-aligned (no regression) while the modal content is centered.
This way StudioPage stays neutral and each consumer controls its own centering needs. Tested both online and offline - the download button is visible and properly aligned in both states.

vtushar06 commented:
Thanks! Really appreciate your feedback throughout this PR.
